insights/combiners/crio_conf.py

```diff
@@ -0,0 +1,150 @@
+"""
+crio configuration
+==================
+The crio files are normally available to rules as a list of CrioConf objects.
+"""
+
+from insights.contrib.ConfigParser import NoOptionError, NoSectionError
+from insights.core.plugins import combiner
+from insights.parsers.crio_conf import CrioConf
+
+
+@combiner(CrioConf)
+class AllCrioConf(object):
+    """
+    Combiner for accessing all the crio configuration files. There may be multi
+    files for crio configuration, and the main config file is crio.conf. In the
+    situation that the same section is both in crio.conf and other configuration
+    files, the item in crio.conf has the lowest precedence. This combiner will
+    parse all the CrioConf objects and return a dictionary containing all valid data.
+
+    Sample files::
+
+        /etc/crio/crio.conf:
+
+            [crio]
+            storage_driver = "overlay"
+            storage_option = [
+                    "overlay.override_kernel_check=1",
+            ]
+
+            [crio.runtime]
+            selinux = true
+
+            [crio.network]
+            plugin_dirs = [
+                    "/usr/libexec/cni",
+            ]
+            [crio.metrics]
+
+        /etc/crio/crio.conf.d/00-conmon.conf
+
+            [crio]
+            internal_wipe = true
+            storage_driver = "device mapper"
+            storage_option = [
+                "overlay.override_kernel_check=1",
+            ]
+
+            [crio.api]
+            stream_address = ""
+            stream_port = "10010"
+
+            [crio.runtime]
+            selinux = true
+            conmon = ""
+            conmon_cgroup = "pod"
+            default_env = [
+                "NSS_SDB_USE_CACHE=no",
+            ]
+            log_level = "info"
+            cgroup_manager = "systemd"
+
+    Examples:
+        >>> all_crio_conf.sections()
+        ['crio', 'crio.runtime', 'crio.api', 'crio.network', 'crio.metrics']
+        >>> all_crio_conf.options('crio.api')
+        ['stream_address', 'stream_port']
+        >>> all_crio_conf.files
+        ['/etc/crio/crio.conf', '/etc/crio/crio.conf.d/00-conmon.conf']
+        >>> all_crio_conf.get('crio', 'storage_driver')
+        '"device mapper"'
+
+    Attributes:
+        files (list): The list of configuration file names.
+    """
+    def __init__(self, crio_confs):
+        self.data = {}
+        self.files = []
+        main_data = None
+        conf_d_data = []
+
+        def dict_merge(dest, src):
+            if not src:
+                return
+
+            for section in src.sections():
+                if section not in dest:
+                    dest[section] = {}
+
+                for option in src.items(section):
+                    dest[section][option] = src.get(section, option)
+
+        for crio_conf in crio_confs:
+            self.files.append(crio_conf.file_path)
+            if crio_conf.file_path == "/etc/crio/crio.conf":
+                main_data = crio_conf
```

Comment:
nitpick: the variable `main_data` is not required, you can merge it to the `self.data` here directly.

Reply:
I removed the redundant main_data. Please have a look.